### ISSUE TITLE
BUG: Missing SIMD typedefs for macOS builds

### ISSUE THREAD

**Issue by nathanjhood** [StoneyDSP/StoneyVCV#4]
# Bug report

Thank you for reporting an issue with StoneyDSP.

If you are looking for help, please try these first:

- Docs: https://docs.StoneyDSP.com
- Discussions: https://github.com/StoneyDSP/StoneyDSP/discussions
- Discord: https://discord.StoneyDSP.com

Before opening a bug report, please verify the following:

- [ x ] I confirm this is a bug with StoneyDSP, not with my own application or with JUCE.
- [ x ] I confirm I have searched the [Docs](https://docs.stoneydsp.com), GitHub [Discussions](https://github.com/StoneyDSP/StoneyDSP/discussions), and [Discord](https://discord.StoneyDSP.com).

## Describe the bug?

CI report for macOS build:

```shell
Run make
c++  -std=c++11 -stdlib=libc++  -Iinclude -Idep/include -fPIC -I/Users/runner/work/StoneyVCV/StoneyVCV/dep/Rack-SDK/include -I/Users/runner/work/StoneyVCV/StoneyVCV/dep/Rack-SDK/dep/include -MMD -MP -g -O3 -funsafe-math-optimizations -fno-omit-frame-pointer -Wall -Wextra -Wno-unused-parameter -march=armv8-a+fp+simd -mmacosx-version-min=10.9  -c -o build/src/plugin.cpp.o src/plugin.cpp
In file included from src/plugin.cpp:34:
include/plugin.hpp:77:3: warning: "No modules found..." [-W#warnings]
 #warning "No modules found..."
  ^
In file included from src/plugin.cpp:41:
In file included from dep/include/../src/StoneyDSP:34:
In file included from dep/include/StoneyDSP.hpp:[17](https://github.com/StoneyDSP/StoneyVCV/actions/runs/11844447208/job/33007690078#step:5:18)7:
dep/include/StoneyDSP/simd.hpp:101:21: error: unknown type name '__m128d_u'; did you mean '__m128d'?
using __UDOUBLE_T = __m128d_u;
                    ^~~~~~~~~
                    __m128d
/Users/runner/work/StoneyVCV/StoneyVCV/dep/Rack-SDK/dep/include/simde/x86/sse2.h:254:24: note: '__m128d' declared here
  typedef simde__m128d __m128d;
                       ^
In file included from src/plugin.cpp:41:
In file included from dep/include/../src/StoneyDSP:34:
In file included from dep/include/StoneyDSP.hpp:177:
dep/include/StoneyDSP/simd.hpp:102:20: error: unknown type name '__m128_u'
using __UFLOAT_T = __m128_u;
                   ^
dep/include/StoneyDSP/simd.hpp:103:[18](https://github.com/StoneyDSP/StoneyVCV/actions/runs/11844447208/job/33007690078#step:5:19): error: unknown type name '__m128i_u'; did you mean '__m128i'?
using __UINT_T = __m128i_u;
                 ^~~~~~~~~
                 __m128i
/Users/runner/work/StoneyVCV/StoneyVCV/dep/Rack-SDK/dep/include/simde/x86/sse2.h:253:24: note: '__m128i' declared here
  typedef simde__m128i __m128i;
                       ^
In file included from src/plugin.cpp:41:
In file included from dep/include/../src/StoneyDSP:34:
In file included from dep/include/StoneyDSP.hpp:177:
dep/include/StoneyDSP/simd.hpp:131:18: error: unknown type name '__UFLOAT_T'; did you mean '__FLOAT_T'?
using ufloat_t = __UFLOAT_T;
                 ^~~~~~~~~~
                 __FLOAT_T
dep/include/StoneyDSP/simd.hpp:98:7: note: '__FLOAT_T' declared here
using __FLOAT_T = __m128;
      ^
dep/include/StoneyDSP/simd.hpp:700:[23](https://github.com/StoneyDSP/StoneyVCV/actions/runs/11844447208/job/33007690078#step:5:24): error: use of undeclared identifier '_mm_loadu_si8'
        return Vector(_mm_loadu_si8(x));
                      ^
dep/include/StoneyDSP/simd.hpp:7[24](https://github.com/StoneyDSP/StoneyVCV/actions/runs/11844447208/job/33007690078#step:5:25):9: error: use of undeclared identifier '_mm_storeu_si8'
        _mm_storeu_si8(x, v);
        ^
src/plugin.cpp:58:6: warning: "No modules found..." [-W#warnings]
    #warning "No modules found..."
     ^
2 warnings and 6 errors generated.
make: *** [build/src/plugin.cpp.o] Error 1
Error: Process completed with exit code 2.
```

## Detailed steps on how to reproduce the bug?

[Here](https://github.com/StoneyDSP/StoneyVCV/actions/runs/11844447208).

## What is the expected behaviour?

Build should pass.

## Screenshots?

See above.

## Operating system(s), version(s), architectures?

macOS x64 + arm64.

## Additional context?

These simd typedefs for 8-bit integers appear to not exist in the case of this particular target platform/arch.

Gladly, SIMD ints are highly interchangeable and we have many operator overloads to help. We may also use `STONEYDSP_MAC` to test and choose an appropriate function definition.

The unsigned floats/doubles are probably not too useful... but I'll try to salvage them if possible.

## Have you tested against the `development` branch?

Yes.

Code of Conduct
By submitting this issue, you agree to follow our [Code of Conduct](https://github.com/StoneyDSP/.github/blob/production/CODE_OF_CONDUCT.md).

**Comment by nathanjhood** [comment]
[Good info here](https://learn.arm.com/learning-paths/cross-platform/intrinsics/intro/) about using SIMD - including the "SIMD Everywhere" library which the Rack SDK depends on and includes internally.

Here are three solutions:

```cpp
// No SIMDE SSE2

#ifdef __SSE2__
  #include <emmintrin.h>
#else
  #ifdef __aarch64__
    #include "sse2neon.h"
  #else
    #warning SSE2 support is not available. Code will not compile
  #endif
#endif
```

```cpp
// Yes SIMDE SSE2

#if defined(__SSE2__) || defined(__aarch64__)
  #define SIMDE_ENABLE_NATIVE_ALIASES
  #include <simde/x86/sse2.h>
#else
  #warning SSE2/NEON support is not available. Code will not compile.
#endif
```

```cpp
// SSE4

#ifdef __SSE4_2__
 #include <nmmintrin.h>
#else
 #define SIMDE_ENABLE_NATIVE_ALIASES
 #include <simde/x86/sse4.2.h>
#endif
```

**Comment by nathanjhood** [comment]
We'll use this for now:

```
#if defined (__SSE2__)
 #include <emmintrin.h>
#else
 #define SIMDE_ENABLE_NATIVE_ALIASES
 #include <simde/x86/sse2.h>
#endif

#if defined (__SSE4_2__)
 #include <nmmintrin.h>
#else
 #define SIMDE_ENABLE_NATIVE_ALIASES
 #include <simde/x86/sse4.2.h>
#endif
```

We can also add [sse2neon](https://github.com/DLTcollab/sse2neon)'s header to the deps folder...

**Comment by nathanjhood** [comment]
The one comment above did fix the CI pipeline.

I have a big rolling PR for the initial unit tests of basic data types - it wasn't supposed to be so big, but the pipelines uncovered lots of gotchas (as they should). The fix is on #1 

I will merge it once the *other* pipeline is fixed.

**Comment by nathanjhood** [comment]
Closing as fixed.
